Server/Controllers/Generation/GenerationAudio.js

```diff
@@ -1,31 +1,43 @@
 const { downloadFile, uploadFile } = require('../MediaManager/S3Manager/S3Manager');
-const { uid } = require('uid');
 const { Errors } = require("../../datas/Errors.js");
 const axios = require('axios');
+const util = require('util');
+const exec = util.promisify(require('child_process').exec);
+const path = require('path')
+
+const fs = require("fs");
+
 
 //TODO Remplir le header quand sera fait
 /**
  * Concat all the Audios files of a project to create the "Final audio"
- * @param { Request } req { body: {  }
+ * @param { Request } req { body: { projectId: string,  audioInfo: List[id: string, timeStamp: float, duration: float]} }
  * @param { Response } res
  * @returns { response to send }
  */
 
 exports.generationAudio = async function (req, res) {
     console.log("Generating a Final Audio...");
-    if (!req.body.projectId)
+    if (!req.body.projectId || !req.body.audioInfo)
         return (res.status(400).send(Errors.BAD_REQUEST_MISSING_INFOS));
     console.log('Generation Audio');
+    const audio_dest = `${process.env.FILES_DIRECTORY}/Audios/${req.body.projectId}`
     const urlSetStatus = `${process.env.BACKEND_URL}/projects/${req.body.projectId}/setStatus`;
     let returnCode = 200;
     let returnMessage = "You successfully generate the audio";
     let returnStatus = "Done";
-    try { 
+    try {
         await axios.post(urlSetStatus, { statusType: 'InProgress', stepType: 'AudioGeneration' });
-
-        // TODO: DL AUDIO FROM s3 --bucketnName, keyName, saveIt = true
-        // MERGE AUDIOS
-        // send audio to s3
+        audioInfo = await sortInput(req.body.audioInfo);
+        console.log(audioInfo)
+        updatedAudioInfo = await getfiles(req.body.projectId, audioInfo);
+        roadGen = await genBlanks(updatedAudioInfo);
+        ad_file = await concatAudios(roadGen, `${audio_dest}-out.mp3`);
+        if (!ad_file || !fs.existsSync(ad_file))
+            throw('Could not generate the audiodescription file')
```

Comment:
Le message sera vraiment gérer ?

Reply:
corrigé dans "clean code: changing throw to new error"